src/sentry_core.c

```diff
@@ -218,6 +231,17 @@ sentry_options_free(sentry_options_t *opts)
     sentry__path_free(opts->handler_path);
     sentry_transport_free(opts->transport);
     sentry__backend_free(opts->backend);
+
+    sentry_attachment_t *next_attachment = opts->attachments;
+    while (next_attachment) {
+        sentry_attachment_t *attachment = next_attachment;
+        next_attachment = attachment->next;
+
+        sentry__path_free(attachment->path);
```

Comment:
_nit_: For consistency, move this into `sentry__attachment_free`? Otherwise, it'll be hard to find this spot when adding fields to attachments in the future.